kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/bytes-simplification.k

```diff
@@ -191,10 +216,34 @@ module BYTES-SIMPLIFICATION [symbolic]
        andBool 0 <=Int S2 andBool 0 <=Int W2 andBool S2 +Int W2 <=Int W1
       [simplification, concrete(W1, S2, W2)]
 
+    rule [range-buf-value]:
+      #range (#buf(W1:Int, X:Int), S2:Int, W2:Int) => #buf(W2, X)
+      requires 0 <=Int X andBool X <Int 2 ^Int (8 *Int W2)
+       andBool 0 <=Int S2 andBool 0 <=Int W2 andBool W1 ==Int S2 +Int W2
+      [simplification, concrete(W1, S2, W2)]
+
     rule [range-padRightToWidth]:
       #range(#padRightToWidth(_, BUF), 0, WIDTH) => BUF
       requires lengthBytes(BUF) ==Int WIDTH [simplification]
 
+    rule [range-eq-check]:
+      BZ +Bytes #range ( B, S, W ) ==K B => true
+      requires 0 <=Int S andBool 0 <=Int W
+      andBool lengthBytes( B ) ==Int S +Int W
+      andBool #asInteger( BZ ) ==Int 0
+      andBool lengthBytes( BZ ) ==Int S
+      andBool #asWord ( B ) <Int 2 ^Int ( 8 *Int W )
+      [simplification, concrete(BZ, S, W), comm]
+
+    rule [range-eq-check-ml]:
+      { BZ +Bytes #range ( B, S, W ) #Equals B } => #Top
+      requires 0 <=Int S andBool 0 <=Int W
+      andBool lengthBytes( B ) ==Int S +Int W
+      andBool #asInteger( BZ ) ==Int 0
+      andBool lengthBytes( BZ ) ==Int S
+      andBool #asWord ( B ) <Int 2 ^Int ( 8 *Int W )
+      [simplification, concrete(BZ, S, W), comm]
```

Comment:
These two, especially the second one, have come up as required for some `#Implies` requests at the end of claim-checking.